group1/comm/oauth2_authentication/client_secrets.json

```diff
@@ -0,0 +1 @@
+{"web":{"client_id":"216454920265-v6i1auidua6oahnd8c92mhl2d1li5cct.apps.googleusercontent.com","project_id":"cs673project","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://accounts.google.com/o/oauth2/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"rzpj48h1IsjywZ0M1qO41NZC","redirect_uris":["http://localhost:8000/communication/oauth2/oauth2callback","http://localhost:8000/communication/oauth2/oauth2callback/filedownload"],"javascript_origins":["http://127.0.0.1:8000"]}}
```

Comment:
I take it this is required for the auth to google (hard coded ATM).  Is this the case?  Is the client ID the ID associated with the users google account?
At some point we'll want this in Gitignore or to store a hash of the client ID in the DB.

Reply:
@danbudris this is required for Google oauth2 authentication.  Technically this shouldn't be checked into source.  Ideally for testing purposes everyone would create their own client_secrets.json through the Google Cloud Console (https://console.cloud.google.com) and just drop them into place in the repository without checking them into source.  In a production environment you could also probably do something similar so these are not checked into source. Storing them in the db is not necessarily a bad idea either if it were a document store or a non-relational database.  For the time being I figured this was more convenient.  When you want to use Google's API's you have to register with them.  You tell them what URL will be accessing their API and what the callback URL is for the purposes of authentication.  Then they provide this file to you.  When creating the authentication request you supply either this whole document, or pieces of it.  They keep track of how many requests (and probably other information) that are made on behalf of the account that is registered to these secrets.  After the last presentation and before we merge all the work from this repository to the main project repository I will remove the contents of this file and leave clear instructions in the documentation so the next developers know what they should do.  After the semester I will also be deactivating the specific request for API access which results in these secrets.